src/query/phrase_query/phrase_scorer.rs

```diff
@@ -192,39 +270,49 @@ impl<TPostings: Postings> PhraseScorer<TPostings> {
             }
             intersection_len = intersection(&mut self.left[..intersection_len], &self.right[..]);
             if intersection_len == 0 {
-                return false;
+                return 0;
             }
         }
 
         self.intersection_docset
             .docset_mut_specialized(self.num_terms - 1)
             .positions(&mut self.right);
-        intersection_exists(&self.left[..intersection_len], &self.right[..])
+        intersection_len
```

Comment:
I understand you are trying to be DRY here, but the contract of the `compute_match*` function is horrible.
This makes the code very difficult to understand.